tradeit-android-sdk/src/main/java/it/trade/android/sdk/model/TradeItLinkedBrokerParcelable.java

```diff
@@ -37,13 +57,91 @@ protected void cache() {
+                    return;
+                }
+                if ((e instanceof NullPointerException) || (e instanceof IllegalArgumentException)) {
+                    // that's likely a bug in the application
+                    Thread.currentThread().getUncaughtExceptionHandler()
+                            .uncaughtException(Thread.currentThread(), e);
+                    return;
+                }
+                if (e instanceof IllegalStateException) {
+                    // that's a bug in RxJava or in a custom operator
+                    Thread.currentThread().getUncaughtExceptionHandler()
+                            .uncaughtException(Thread.currentThread(), e);
+                    return;
+                }
+                Log.w(TAG, "Undeliverable exception received, not sure what to do", e);
+            }
+        });
+        Observable.fromArray(this.accounts)
+                .observeOn(AndroidSchedulers.mainThread(), true)
+                .subscribeOn(Schedulers.io())
+                .flatMapIterable(new Function<List<TradeItLinkedBrokerAccountParcelable>, Iterable<TradeItLinkedBrokerAccountParcelable>>() {
+                    @Override
+                    public Iterable<TradeItLinkedBrokerAccountParcelable> apply(@NonNull List<TradeItLinkedBrokerAccountParcelable> linkedBrokerAccountParcelables) throws Exception {
+                        return linkedBrokerAccountParcelables;
+                    }
+                }).map(new Function<TradeItLinkedBrokerAccountParcelable, Single<TradeItLinkedBrokerAccountParcelable>>() {
+                    @Override
+                    public Single<TradeItLinkedBrokerAccountParcelable> apply(@NonNull final TradeItLinkedBrokerAccountParcelable linkedBrokerAccountParcelable) throws Exception {
+                        return Single.create(new SingleOnSubscribe<TradeItLinkedBrokerAccountParcelable>() {
+                            @Override
+                            public void subscribe(@NonNull final SingleEmitter<TradeItLinkedBrokerAccountParcelable> emmiter) throws Exception {
+                                linkedBrokerAccountParcelable.refreshBalance(new TradeItCallback<TradeItBalanceParcelable>() {
+                                    @Override
+                                    public void onSuccess(TradeItBalanceParcelable balance) {
+                                        emmiter.onSuccess(linkedBrokerAccountParcelable);
+                                    }
+
+                                    @Override
+                                    public void onError(TradeItErrorResult error) {
+                                        Log.e(TAG, error.toString());
+                                        emmiter.onError(new RuntimeException(error.toString()));
+                                    }
+                                });
+                            }
+                        });
```

Comment:
Can we move `.toObservable()` to the end of the `Single.create(new SingleOnSubscribe<TradeItLinkedBrokerAccountParcelable>() { ... })` so that we don't need to have an extra `flatMap` below? Same for `authenticateAll`.